azurecompute/src/main/java/org/jclouds/azurecompute/features/StorageAccountApi.java

```diff
@@ -0,0 +1,98 @@
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.features;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_XML;
+import java.util.List;
+
+import javax.inject.Named;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import org.jclouds.Fallbacks;
+import org.jclouds.Fallbacks.NullOnNotFoundOr404;
+import org.jclouds.azurecompute.binders.StorageServiceParamsToXML;
+import org.jclouds.azurecompute.domain.Availability;
+import org.jclouds.azurecompute.domain.StorageService;
+import org.jclouds.azurecompute.domain.StorageServiceParams;
+import org.jclouds.azurecompute.functions.ParseRequestIdHeader;
+import org.jclouds.azurecompute.xml.AvailabilityHandler;
+import org.jclouds.azurecompute.xml.ListStorageServicesHandler;
+import org.jclouds.azurecompute.xml.StorageServiceHandler;
+import org.jclouds.rest.annotations.BinderParam;
+import org.jclouds.rest.annotations.Fallback;
+import org.jclouds.rest.annotations.Headers;
+import org.jclouds.rest.annotations.ResponseParser;
+import org.jclouds.rest.annotations.XMLResponseParser;
+
+@Path("/services/storageservices")
+@Headers(keys = "x-ms-version", values = "{jclouds.api-version}")
+@Consumes(MediaType.APPLICATION_XML)
+public interface StorageAccountApi {
+
+   /**
+    * The Create Storage Account asynchronous operation creates a new storage account in Microsoft Azure.
+    *
+    */
+   @Named("CreateStorageAccount")
+   @POST
+   @Produces(APPLICATION_XML)
+   @ResponseParser(ParseRequestIdHeader.class)
+   String create(@BinderParam(StorageServiceParamsToXML.class) StorageServiceParams storageServiceParams);
+
+   /**
+    * https://management.core.windows.net/<subscription-id>/services/storageservices
+    */
+   @Named("GetStorageAccountDetails")
+   @GET
+   @Path("/{storageAccountName}")
+   @XMLResponseParser(StorageServiceHandler.class)
+   @Fallback(Fallbacks.VoidOnNotFoundOr404.class)
+   StorageService get(@PathParam("subscriptionId") String subscriptionId, @PathParam("storageAccountName") String storageAccountName);
+
+   /**
+    * The List Storage Accounts operation lists the storage accounts that are available in the specified subscription.
+    */
+   @Named("ListStorageAccounts")
+   @GET
+   @XMLResponseParser(ListStorageServicesHandler.class)
+   @Fallback(Fallbacks.EmptyListOnNotFoundOr404.class)
+   List<StorageService> list(@PathParam("subscriptionId") String subscriptionId);
+
+   /**
+    * The Check Storage Account Name Availability operation checks to see if the specified storage account name is available, or if it has already been taken.
+    */
+   @Named("CheckStorageAccountNameAvailability")
+   @GET
+   @Path("/operations/isavailable/{storageAccountName}")
+   @XMLResponseParser(AvailabilityHandler.class)
+   @Fallback(NullOnNotFoundOr404.class)
```

Comment:
What does it mean that a check returns `null`? Shouldn't it return just a boolean? if you want to return an object, shouldn't it be better to throw an exception if the check can not be performed? (I think it is not clear what a null means in this case: not available/the check operation failed).

Consider removing the fallback to propagate an exception.